src/db.rs

```diff
@@ -30,7 +30,22 @@ pub struct CacheEntry {
 }
 
 fn is_compatible(debversion: &str, crateversion: &VersionReq) -> Result<bool, Error> {
-    let debversion = debversion.replace('~', "-");
+    let mut debversion = debversion.replace('~', "-");
+    if debversion.contains('+') {
+        let parts: Vec<&str> = debversion.split('+').collect();
+        debversion = match parts[0].matches('.').count() {
+            0 => {
+                format!("{}.0.0", parts[0])
+            }
+            1 => {
+                format!("{}.0", parts[0])
+            }
+            2 => parts[0].to_owned(),
+            _ => {
+                return Err(anyhow!("wrong number of '.' characters in semver string"));
+            }
+        };
+    }
```

Comment:
It's possible to avoid `parts[0]` entirely by matching on the return value of `.split_once`:

```suggestion
    if let Some((version, _suffix)) = debversion.split_once('+') {
        debversion = match version.matches('.').count() {
            0 => format!("{version}.0.0"),
            1 => format!("{version}.0"),
            2 => version.to_owned(),
            _ => bail!("wrong number of '.' characters in semver string: {version:?}")),
        };
    }
```

Also adding the version string into the error message to make it easier to debug. :)

Reply:
great suggestions, thanks :)